FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,317 @@
+
+namespace FGEGraphics.GraphicsHelpers.Models;
+
+/// <summary>Generates shapes and returns them as a model.</summary>
+public static class ShapeGenerators
+{
+    /// <summary>
+    /// Generates a 3D sphere model.
+    /// Note that the stacks should be lower than the slices to avoid incorrect normals a good example is 10 stacks and 60 slices.
+    /// </summary>
+    public static Model GenerateSphere(float radius, uint stacks, uint slices, ModelEngine modelEngine, bool reverseOrder = false)
+    {
+        if (stacks > slices)
+        {
+            Logs.Warning("Sphere has more stacks than slices, this may result in incorrect normals.");
+        }
+
+        uint vertexCount = (stacks + 1) * (slices + 1);
+        uint numIndices = stacks * slices * 6;
+
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+
+        for (int i = 0; i <= stacks; i++)
+        {
+            float theta = i * MathHelper.Pi / stacks;
+            float sinTheta = (float)Math.Sin(theta);
+            float cosTheta = (float)Math.Cos(theta);
+
+            for (int j = 0; j <= slices; j++)
+            {
+                float phi = j * 2 * MathHelper.Pi / slices;
+                float sinPhi = (float)Math.Sin(phi);
+                float cosPhi = (float)Math.Cos(phi);
+
+                float x = radius * cosPhi * sinTheta;
+                float y = radius * cosTheta;
+                float z = radius * sinPhi * sinTheta;
+
+                vertices.Add(new Vector3(x, y, z));
+                normals.Add(new Vector3(x, y, z).Normalized());
+                texCoords.Add(new Vector2((float)j / slices, (float)i / stacks));
+
+                if (i < stacks && j < slices)
+                {
+                    uint currentRow = (uint)(i * (slices + 1));
+                    uint nextRow = (uint)((i + 1) * (slices + 1));
+
+                    indices[index++] = (uint)(currentRow + j);
+                    indices[index++] = (uint)(nextRow + j);
+                    indices[index++] = (uint)(currentRow + j + 1);
+
+                    indices[index++] = (uint)(currentRow + j + 1);
+                    indices[index++] = (uint)(nextRow + j);
+                    indices[index++] = (uint)(nextRow + j + 1);
+                }
+            }
+        }
+        if (reverseOrder)
+        {
+            Array.Reverse(indices);
+        }
+        return GetModelAfterGenerating(modelEngine, "sphere", vertices, normals, texCoords, indices);
+    }
+
+    /// <summary>Generates a 2D circle model.</summary>
+    public static Model Generated2DCircle(float radius, uint slices, ModelEngine modelEngine, bool reverseOrder = false)
+    {
+        uint vertexCount = slices;
+        uint numIndices = slices * 3;
+
+        List<Vector3> vertices = new((int)vertexCount);
```

Comment:
this code would be a lot cleaner if the params were `int`s in the first place, rather than `uint` inputs that have to be cast all over the place